src/options/set.rs

```diff
@@ -556,14 +556,31 @@ fn set_widths(
         background_color_extends_to_terminal_width;
 }
 
-fn set_true_color(opt: &mut cli::Opt) {
+fn set_true_color(
+    opt: &mut cli::Opt,
+    git_config: &mut Option<GitConfig>,
+    arg_matches: &clap::ArgMatches,
+    option_names: &HashMap<&str, &str>,
+) {
     if opt.true_color == "auto" {
         // It's equal to its default, so the user might be using the deprecated
         // --24-bit-color option.
         if let Some(_24_bit_color) = opt._24_bit_color.as_ref() {
             opt.true_color = _24_bit_color.clone();
+        } else {
+            let empty_builtin_features = HashMap::new();
+            set_options!(
+                [true_color],
+                opt,
+                &empty_builtin_features,
+                git_config,
+                arg_matches,
+                option_names,
+                false
+            );
```

Comment:
`true_color` is present in the main call to `set_options!` on [l.182](https://github.com/dandavison/delta/blob/93e081985fe7192f7cb6f472ae8d6fb835cdc0e4/src/options/set.rs#L181-L182). I think we need to know why that call is not working -- it feels wrong to make two `set_options!` calls for `true_color`.

Reply:
Oh I see, and sorry. I missed that and found `opt.true_color` is set correctly from gitconfig after set_options. I'm maybe going to fix.